src/main/ruby/core/truffle/cext.rb

```diff
@@ -78,11 +78,17 @@ def self.library_in_search_path_or_default(lib, search_paths)
     end
 
     def self.standardize_lib_name(lib)
-      if lib.start_with?('lib') or lib.start_with?('/')
-        lib
-      else
-        "lib#{lib}.#{RbConfig::CONFIG['NATIVE_DLEXT']}"
-      end
+      require 'ffi'
+      extend FFI::Library
+
+      mapped_lib = FFI.map_library_name(lib)
+
+      # Mapping a library name will only handle hard-coded cases accurately. Otherwise, it takes a best effort approach
+      # towards standardizing a library name. On Linux, that name may correspond to an `ld` script, rather than the
+      # actual library we want to load. So, we try loading the library here, which will properly resolve the full
+      # library path if necessary.
+      dynamic_lib = ffi_lib(mapped_lib).first
```

Comment:
Is it needed to call `map_library_name` first or can we let `ffi_lib` do it?

Reply:
I think so. I'll have to verify if `dlopen` is smart enough to handle short names like "m". `map_iibrary_name` will convert "m" to "libm.so". It does the same thing as the code it replaced.

Reply:
I see, I think it makes sense to keep the explicit `map_iibrary_name` first then, could you add an example in a comment above it to clarify what it does?
I think C compilers basically do a similar transformation when using `-l`, i.e. it's not dlopen()'s job.